webapp/data_structures/EXPRESSION.js

```diff
@@ -0,0 +1,114 @@
+function EXPRESSION () {
+
+    EXPRESSION.prototype.define = function EXPRESSION() {
+        this.in = [[1], [1]];
+        this.out = [[1]];
+        this.exx = "(u1>0)*sin(u2)^2";
+        this.usenz = 1;
+        var model = scicos_model();
+        model.sim = list(new ScilabString(["evaluate_expr"]), new ScilabDouble([4]));
+        model.in = new ScilabDouble(...this.in);
+        model.out = new ScilabDouble([this.out]);
+	    model.rpar = new ScilabDouble([0], [2]);
+	    model.ipar = new ScilabDouble([2], [1], [6], [1], [5], [18], [2], [2], [5], [101], [6], [2], [5], [15], [5], [3]);
+	    model.nzcross = new ScilabDouble([[1]]);
+	    model.nmode = new ScilabDouble([1]);
+	    model.dep_ut = new ScilabBoolean([true,false]);
+	    this.in = (size(this.in,"*"));
+	    var exprs = new ScilabString([this.in],[this.exx],[this.usenz]);
+	    this.displayParameter = [[this.exx]];
+	    var gr_i = [];
+	    this.x = new standard_define(new ScilabDouble([5,2]),model,exprs,gr_i);
+	    return new BasicBlock(this.x)
+    }
+    EXPRESSION.prototype.details = function EXPRESSION() {
+        return this.x;
+    }
+    EXPRESSION.prototype.set = function EXPRESSION() {
+        this.in = inverse(arguments[0]["in"] || arguments[0]["inp"])
+        this.exx = arguments[0]["exx"]
+        this.usenz = parseFloat((arguments[0]["usenz"]))
+
+        if(this.exx.length == 0){
+            this.exx = "0";
+        }
+        if(this.in != 0){
+            if(this.in == 1){
+                this.nini = 8;
+            }else{
+                this.nini = this.in;
+            }
+        }else{
+            alert("Variable u1 is not defined.");
+            EXPRESSION.get();
+        }
+
+        this.head = "%foo(";
+        for(var i = 1; i <= this.nini-1; i++){
+
+            this.head = this.head + "u" + i.toString() + ",";
+
+        }
+        this.head = this.head + "u" + this.nini.toString() + ")";
+        var get_values = get_expr_output_for_DefineandSet(this.head,this.exx);
+        this.rpar = JSON.parse(get_values.rpar);
+        this.ipar = JSON.parse(get_values.ipar);
+        this.nz = JSON.parse(get_values.nz);
+        var ok = get_values.ok;
+        var ok1 = get_values.ok1;
+        if(ok == "false"){
+            alert("Erroneous expression "+output);
+            EXPRESSION.get();
+        }else{
+            if(this.in > 1){
+                this.inp = []
+                for (var i = 1; i <= this.in; i++ ) {
+                    this.inp.push([-1*i])
+                }
+                var io = check_io(this.x.model,this.x.graphics,this.inp,[1],[],[])
+            }else{
+                var io = check_io(this.x.model,this.x.graphics,[-1],[1],[],[])
+            }
+            if(ok == "false"){
+                alert("Erroneous expression "+output);
+                EXPRESSION.get();
```

Comment:
```suggestion
                return false;
```

return a value here.